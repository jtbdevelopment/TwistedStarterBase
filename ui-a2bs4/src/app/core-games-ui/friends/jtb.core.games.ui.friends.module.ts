import {NgModule} from '@angular/core';
import {HttpModule} from '@angular/http';
import {JTBCoreGamesUIMessageBus} from '../messagebus/jtb.core.games.ui.messagebus.module';
import {FriendsService} from './friends.service';

@NgModule({
    imports: [HttpModule, JTBCoreGamesUIMessageBus],
    exports: [JTBCoreGamesUIMessageBus],
    providers: [
        FriendsService
    ]
})
export class JTBCoreGamesUIFriends {
}
