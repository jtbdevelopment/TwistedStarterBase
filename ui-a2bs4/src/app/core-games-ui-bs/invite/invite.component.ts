import {Component, Inject} from '@angular/core';
import {FacebookInviteService} from '../../core-games-ui/facebook/facebook-invite.service';
import {NgbActiveModal} from '@ng-bootstrap/ng-bootstrap';
import {AppConfig} from '../../core-games-ui/appconfig.interface';
import {FriendsService} from '../../core-games-ui/friends/friends.service';
import {Invitable} from '../../core-games-ui/friends/invitable.model';

@Component({
    selector: 'invite-friends',
    template: require('./invite.component.html'),
})
export class InviteComponent {
    public invitable: Invitable[] = [];
    public chosen: Invitable[] = [];

    constructor(private facebookInviteService: FacebookInviteService,
                private friendService: FriendsService,
                private modalRef: NgbActiveModal,
                @Inject('AppConfig') private config: AppConfig) {
        this.friendService.invitableFriends.subscribe(f => {
            this.invitable = f;
            this.chosen = [];
        });
    }

    public cancel(): void {
        this.modalRef.dismiss();
    }

    public inviteFriends(): void {
        this.facebookInviteService.inviteFriends(this.chosen, this.config.inviteFriendsMessage);
        this.modalRef.close();
    }
}
