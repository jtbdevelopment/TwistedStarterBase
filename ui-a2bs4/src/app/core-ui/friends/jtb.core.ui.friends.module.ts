import {NgModule} from '@angular/core';
import {HttpModule} from '@angular/http';
import {JTBCoreUIMessageBus} from '../messagebus/jtb.core.ui.messagebus.module';
import {FriendsService} from './friends.service';

@NgModule({
    imports: [HttpModule, JTBCoreUIMessageBus],
    exports: [JTBCoreUIMessageBus],
    providers: [
        FriendsService
    ]
})
export class JTBCoreUIFriends {
}
