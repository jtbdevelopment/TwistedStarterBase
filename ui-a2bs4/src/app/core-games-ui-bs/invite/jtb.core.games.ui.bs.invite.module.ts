import {NgModule} from '@angular/core';
import {BrowserModule} from '@angular/platform-browser';
import {NgbModule} from '@ng-bootstrap/ng-bootstrap';
import {JTBCoreGamesUI} from '../../core-games-ui/jtb.core.games.ui.module';
import {InviteComponent} from './invite.component';
import {MultiSelectModule} from 'primeng/primeng';
import {FormsModule} from '@angular/forms';

@NgModule({
    imports: [
        BrowserModule,
        NgbModule,
        FormsModule,
        MultiSelectModule,
        JTBCoreGamesUI
    ],
    providers: [
        {provide: 'Window', useValue: window}
    ],
    entryComponents: [
        InviteComponent
    ],
    declarations: [
        InviteComponent
    ]
})
export class JTBCoreGamesUIBInviteModule {
}

