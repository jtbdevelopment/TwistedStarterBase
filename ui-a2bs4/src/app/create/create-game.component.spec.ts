import {fakeAsync, TestBed, tick} from '@angular/core/testing';
import {NgbModal, NgbModule, NgbTabsetConfig, NgbTooltipConfig} from '@ng-bootstrap/ng-bootstrap';
import {CreateGameComponent} from './create-game.component';
import {FormsModule} from '@angular/forms';
import {FeatureGroup} from '../core-games-ui/features/feature-group.model';
import {BehaviorSubject} from 'rxjs/BehaviorSubject';
import {FeatureCacheService} from '../core-games-ui/features/feature-cache.service';
import {Feature} from '../core-games-ui/features/feature.model';
import {FeatureOption} from '../core-games-ui/features/feature-option.model';
import {Friend} from '../core-games-ui/friends/friend.model';
import {Invitable} from '../core-games-ui/friends/invitable.model';
import {FriendsService} from '../core-games-ui/friends/friends.service';
import {BootstrapActionsService} from '../core-games-ui-bs/actions/bootstrap-actions.service';
import {MultiSelectModule} from 'primeng/primeng';
import {InviteComponent} from '../core-games-ui-bs/invite/invite.component';

class MockFeatureService {
    public features: BehaviorSubject<FeatureGroup[]> = new BehaviorSubject<FeatureGroup[]>([]);
}

class MockFriendService {
    public friends: BehaviorSubject<Friend[]> = new BehaviorSubject<Friend[]>([]);
    public invitableFriends: BehaviorSubject<Invitable[]> = new BehaviorSubject<Invitable[]>([]);

    refreshFriends = jasmine.createSpy('rf');
}

class MockBoostrapActions {
    newGame = jasmine.createSpy('newGame');
}

let groups: FeatureGroup[] = [
    new FeatureGroup('Group1'),
    new FeatureGroup('Group2')
];
groups[0].features = [
    new Feature('feature1', 'feature 1', 'feature 1 desc'),
    new Feature('feature2', 'feature 2', 'feature 2 desc'),
];
groups[1].features = [
    new Feature('feature3', 'feature 3', 'feature 3 desc'),
];
groups[0].features[0].options = [
    new FeatureOption('option1-1', 'option 1-1', 'option 1-1 desc'),
    new FeatureOption('option1-2', 'option 1-2', 'option 1-2 desc'),
    new FeatureOption('option1-3', 'option 1-3', 'option 1-3 desc'),
];
groups[0].features[1].options = [
    new FeatureOption('option2-1', 'option 2-1', 'option 2-1 desc'),
    new FeatureOption('option2-2', 'option 2-2', 'option 2-2 desc'),
];
groups[1].features[0].options = [
    new FeatureOption('option3-1', 'option 3-1', 'option 3-1 desc'),
    new FeatureOption('option3-2', 'option 3-2', 'option 3-2 desc'),
    new FeatureOption('option3-3', 'option 3-3', 'option 3-3 desc'),
    new FeatureOption('option4-3', 'option 4-3', 'option 4-3 desc'),
];

class MockModal {
    public open = jasmine.createSpy('open');
}
describe('Component:  create game component', () => {
    let featureService: MockFeatureService;
    let friendService: MockFriendService;
    let actionService: MockBoostrapActions;
    let modalService: MockModal;

    beforeEach(() => {
        TestBed.configureTestingModule({
            imports: [
                NgbModule,
                FormsModule,
                MultiSelectModule
            ],
            declarations: [
                CreateGameComponent
            ],
            providers: [
                {provide: FeatureCacheService, useClass: MockFeatureService},
                {provide: FriendsService, useClass: MockFriendService},
                {provide: BootstrapActionsService, useClass: MockBoostrapActions},
                {provide: NgbModal, useClass: MockModal},
                NgbTabsetConfig,
                NgbTooltipConfig
            ]
        });
        TestBed.compileComponents();
        featureService = TestBed.get(FeatureCacheService);
        friendService = TestBed.get(FriendsService);
        actionService = TestBed.get(BootstrapActionsService) as MockBoostrapActions;
        modalService = TestBed.get(NgbModal);
    });

    it('initializes', () => {
        const fixture = TestBed.createComponent(CreateGameComponent);

        fixture.detectChanges();
        let empty = {};
        expect(fixture.componentInstance.choices).toEqual(empty);
        expect(fixture.componentInstance.groups).toEqual([]);
        expect(fixture.componentInstance.friends).toEqual([]);
        expect(fixture.componentInstance.disableCreate).toEqual(true);
    });

    it('subscribes to game features', () => {
        const fixture = TestBed.createComponent(CreateGameComponent);

        expect(fixture.componentInstance.disableCreate).toEqual(true);
        featureService.features.next(groups);
        fixture.detectChanges();

        expect(fixture.componentInstance.disableCreate).toEqual(false);
        expect(fixture.componentInstance.groups).toEqual(groups);
        expect(JSON.stringify(fixture.componentInstance.choices)).toEqual(JSON.stringify(
            {
                feature1: 'option1-1',
                feature2: 'option2-1',
                feature3: 'option3-1',
            }
        ));


        groups.forEach(group => {
            let tab = fixture.nativeElement.querySelector('#' + group.groupType);
            expect(tab).toBeDefined();
        });
    });

    it('subscribes to friends', () => {
        const fixture = TestBed.createComponent(CreateGameComponent);
        expect(friendService.refreshFriends).toHaveBeenCalledTimes(1);
        let newFriend: Friend[] = [new Friend('md5', 'dn1'), new Friend('md', 'dn')];
        let newInt: Invitable[] = [new Invitable('id1', '1')];
        friendService.friends.next(newFriend);
        friendService.invitableFriends.next(newInt);
        expect(fixture.componentInstance.friends).toEqual(newFriend);
    });

    describe('after initialized', () => {
        let fixture;
        beforeEach(fakeAsync(() => {
            fixture = TestBed.createComponent(CreateGameComponent);
            featureService.features.next(groups);
            tick();
            fixture.detectChanges();
        }));

        it('invite friends opens dialog', () => {
            fixture.nativeElement.querySelector('.invite-friends-button').click();
            expect(modalService.open).toHaveBeenCalledWith(InviteComponent);
        });
        
        it('submits a solo game', () => {
            fixture.nativeElement.querySelector('#option2-2').click();
            fixture.nativeElement.querySelector('#option1-3').click();
            fixture.detectChanges();

            expect(JSON.stringify(fixture.componentInstance.choices)).toEqual(JSON.stringify(
                {
                    feature1: 'option1-3',
                    feature2: 'option2-2',
                    feature3: 'option3-1',
                }
            ));

            fixture.nativeElement.querySelector('.create-game-button').click();
            expect(actionService.newGame).toHaveBeenCalledWith({
                players: [],
                features: ['option1-3', 'option2-2', 'option3-1']
            });
        });

        it('submits a multi-player game', () => {
            //  Not testing clicking via prime-ng component - assume it works
            fixture.nativeElement.querySelector('#option2-2').click();
            fixture.nativeElement.querySelector('#option1-3').click();
            fixture.detectChanges();
            fixture.componentInstance.chosenFriends = [new Friend('md51', 'dn1'), new Friend('md52', 'dn2')];


            expect(JSON.stringify(fixture.componentInstance.choices)).toEqual(JSON.stringify(
                {
                    feature1: 'option1-3',
                    feature2: 'option2-2',
                    feature3: 'option3-1',
                }
            ));

            fixture.nativeElement.querySelector('.create-game-button').click();
            expect(actionService.newGame).toHaveBeenCalledWith({
                players: ['md51', 'md52'],
                features: ['option1-3', 'option2-2', 'option3-1']
            });
        });
    });
});
