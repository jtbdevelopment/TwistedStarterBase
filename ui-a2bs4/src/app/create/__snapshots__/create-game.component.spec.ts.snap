// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component:  create game component initializes 1`] = `
<app-create-game
  choices={[Function Object]}
  chosenFriends={[Function Array]}
  createGameText={[Function String]}
  disableCreate={[Function Boolean]}
  featureCache={[Function MockFeatureService]}
  friendService={[Function MockFriendService]}
  friends={[Function Array]}
  gameActions={[Function MockBoostrapActions]}
  groups={[Function Array]}
  ngbModal={[Function MockModal]}
>
  <div
    class="create-game"
  >
    
    
    <form
      class="ng-untouched ng-pristine ng-valid"
      novalidate=""
    >
      
      
      
      
      <ngb-tabset
        ng-reflect-type="pills"
        type="pills"
      >
        
      
        <ul
          class="nav nav-pills justify-content-start"
          role="tablist"
        >
          
        
          
          
      
        </ul>
        
      
        <div
          class="tab-content"
        >
          
        
          
          
      
        </div>
        
    
      </ngb-tabset>
      
      
      <div
        class="row"
      >
        
        
        <label
          class="option-label col-md-2"
        >
          Opponents
        </label>
        
        
        <div
          class="col-md-10"
        >
          
          
          <p-multiselect
            class="ui-inputwrapper-filled ng-untouched ng-pristine ng-valid"
            datakey="md5"
            maxselectedlabels="10"
            ng-reflect-data-key="md5"
            ng-reflect-max-selected-labels="10"
            ng-reflect-model=""
            ng-reflect-option-label="displayName"
            ng-reflect-options="[object Object]"
            optionlabel="displayName"
          >
            
          
            <div
              class="ui-multiselect ui-widget ui-state-default ui-corner-all"
              ng-reflect-ng-class="[object Object]"
            >
              
              
              <div
                class="ui-helper-hidden-accessible"
              >
                
                  
                <input
                  readonly="readonly"
                  type="text"
                />
                
              
              </div>
              
              
              <div
                class="ui-multiselect-label-container"
                title="Choose"
              >
                
                  
                <label
                  class="ui-multiselect-label ui-corner-all"
                >
                  Choose
                </label>
                
              
              </div>
              
              
              <div
                class="ui-multiselect-trigger ui-state-default ui-corner-right"
                ng-reflect-ng-class="[object Object]"
              >
                
                  
                <span
                  class="ui-clickable fa fa-fw fa-caret-down"
                  ng-reflect-klass="ui-clickable"
                  ng-reflect-ng-class="fa fa-fw fa-caret-down"
                />
                
              
              </div>
              
              
              <div
                class="ui-multiselect-panel ui-widget ui-widget-content ui-corner-all ui-shadow"
                ng-reflect-ng-class="ui-multiselect-panel ui-widget"
                style="display: none;"
              >
                
                  
                <div
                  class="ui-widget-header ui-corner-all ui-multiselect-header ui-helper-clearfix"
                  ng-reflect-klass="ui-widget-header ui-corner-all"
                  ng-reflect-ng-class="[object Object]"
                >
                  
                      
                  
                  <div
                    class="ui-chkbox ui-widget"
                  >
                    
                          
                    <div
                      class="ui-helper-hidden-accessible"
                    >
                      
                              
                      <input
                        readonly="readonly"
                        type="checkbox"
                      />
                      
                          
                    </div>
                    
                          
                    <div
                      class="ui-chkbox-box ui-widget ui-corner-all ui-state-default"
                      ng-reflect-klass="ui-chkbox-box ui-widget ui-cor"
                      ng-reflect-ng-class="[object Object]"
                    >
                      
                              
                      <span
                        class="ui-chkbox-icon ui-clickable"
                        ng-reflect-klass="ui-chkbox-icon ui-clickable"
                        ng-reflect-ng-class="[object Object]"
                      />
                      
                          
                    </div>
                    
                      
                  </div>
                  
                      
                  
                  <div
                    class="ui-multiselect-filter-container"
                  >
                    
                          
                    <input
                      class="ui-inputtext ui-widget ui-state-default ui-corner-all"
                      role="textbox"
                      type="text"
                    />
                    
                          
                    <span
                      class="fa fa-fw fa-search"
                    />
                    
                      
                  </div>
                  
                      
                  <a
                    class="ui-multiselect-close ui-corner-all"
                    href="#"
                  >
                    
                          
                    <span
                      class="fa fa-close"
                    />
                    
                      
                  </a>
                  
                  
                </div>
                
                  
                <div
                  class="ui-multiselect-items-wrapper"
                >
                  
                      
                  <ul
                    class="ui-multiselect-items ui-multiselect-list ui-widget-content ui-widget ui-corner-all ui-helper-reset"
                    style="max-height: 200px;"
                  >
                    
                          
                    
                    
                      
                  </ul>
                  
                  
                </div>
                
              
              </div>
              
          
            </div>
            
      
          </p-multiselect>
          
        
        </div>
        
      
      </div>
      
      
      <div
        class="row"
      >
        
        
        <div
          class="col-md-6"
        >
          
          
          <button
            class="btn btn-primary create-game-button w-100"
            disabled=""
          >
            
            Create Game!
          
          </button>
          
        
        </div>
        
        
        <div
          class="col-md-6"
        >
          
          
          <button
            class="btn btn-info invite-friends-button w-100"
          >
            
            Invite friends to play
          
          </button>
          
        
        </div>
        
      
      </div>
      
    
    </form>
    
  
  </div>
  
</app-create-game>
`;

exports[`Component:  create game component subscribes to friends 1`] = `
<app-create-game
  choices={[Function Object]}
  chosenFriends={[Function Array]}
  createGameText={[Function String]}
  disableCreate={[Function Boolean]}
  featureCache={[Function MockFeatureService]}
  friendService={[Function MockFriendService]}
  friends={[Function Array]}
  gameActions={[Function MockBoostrapActions]}
  groups={[Function Array]}
  ngbModal={[Function MockModal]}
>
  <div
    class="create-game"
  >
    
    
    <form
      novalidate=""
    >
      
      
      
      
      <ngb-tabset
        type="pills"
      >
        
      
        <ul
          role="tablist"
        >
          
        
          
          
      
        </ul>
        
      
        <div
          class="tab-content"
        >
          
        
          
          
      
        </div>
        
    
      </ngb-tabset>
      
      
      <div
        class="row"
      >
        
        
        <label
          class="option-label col-md-2"
        >
          Opponents
        </label>
        
        
        <div
          class="col-md-10"
        >
          
          
          <p-multiselect
            datakey="md5"
            maxselectedlabels="10"
            optionlabel="displayName"
          >
            
          
            <div>
              
              
              <div
                class="ui-helper-hidden-accessible"
              >
                
                  
                <input
                  readonly="readonly"
                  type="text"
                />
                
              
              </div>
              
              
              <div
                class="ui-multiselect-label-container"
              >
                
                  
                <label
                  class="ui-multiselect-label ui-corner-all"
                >
                  
                </label>
                
              
              </div>
              
              
              <div>
                
                  
                <span
                  class="ui-clickable"
                />
                
              
              </div>
              
              
              <div>
                
                  
                <div
                  class="ui-widget-header ui-corner-all ui-multiselect-header ui-helper-clearfix"
                >
                  
                      
                  
                  
                      
                  
                  
                      
                  <a
                    class="ui-multiselect-close ui-corner-all"
                    href="#"
                  >
                    
                          
                    <span
                      class="fa fa-close"
                    />
                    
                      
                  </a>
                  
                  
                </div>
                
                  
                <div
                  class="ui-multiselect-items-wrapper"
                >
                  
                      
                  <ul
                    class="ui-multiselect-items ui-multiselect-list ui-widget-content ui-widget ui-corner-all ui-helper-reset"
                  >
                    
                          
                    
                    
                      
                  </ul>
                  
                  
                </div>
                
              
              </div>
              
          
            </div>
            
      
          </p-multiselect>
          
        
        </div>
        
      
      </div>
      
      
      <div
        class="row"
      >
        
        
        <div
          class="col-md-6"
        >
          
          
          <button
            class="btn btn-primary create-game-button w-100"
          >
            
            
          </button>
          
        
        </div>
        
        
        <div
          class="col-md-6"
        >
          
          
          <button
            class="btn btn-info invite-friends-button w-100"
          >
            
            Invite friends to play
          
          </button>
          
        
        </div>
        
      
      </div>
      
    
    </form>
    
  
  </div>
  
</app-create-game>
`;

exports[`Component:  create game component subscribes to game features 1`] = `
<app-create-game
  choices={[Function Object]}
  chosenFriends={[Function Array]}
  createGameText={[Function String]}
  disableCreate="false"
  featureCache={[Function MockFeatureService]}
  friendService={[Function MockFriendService]}
  friends={[Function Array]}
  gameActions={[Function MockBoostrapActions]}
  groups={[Function Array]}
  ngbModal={[Function MockModal]}
>
  <div
    class="create-game"
  >
    
    
    <form
      class="ng-untouched ng-pristine ng-valid"
      novalidate=""
    >
      
      
      
      
      <ngb-tabset
        ng-reflect-type="pills"
        type="pills"
      >
        
      
        <ul
          class="nav nav-pills justify-content-start"
          role="tablist"
        >
          
        
          
          <li
            class="nav-item"
          >
            
          
            <a
              aria-controls="Group1-panel"
              aria-disabled="false"
              aria-expanded="true"
              class="nav-link active"
              href=""
              id="Group1"
              role="tab"
            >
              
            Group1
              
              
          
            </a>
            
        
          </li>
          <li
            class="nav-item"
          >
            
          
            <a
              aria-disabled="false"
              aria-expanded="false"
              class="nav-link"
              href=""
              id="Group2"
              role="tab"
            >
              
            Group2
              
              
          
            </a>
            
        
          </li>
          
      
        </ul>
        
      
        <div
          class="tab-content"
        >
          
        
          
          
          
          
          <div
            aria-expanded="true"
            aria-labelledby="Group1"
            class="tab-pane active"
            id="Group1-panel"
            role="tabpanel"
          >
            
            
            
            
            
            
            <div
              class="form-group row"
            >
              
              
              <label
                class="option-label col-md-2"
                ng-reflect-ngb-tooltip="feature 1 desc"
              >
                feature 1
              </label>
              
              
              <div
                class="col-md-10"
              >
                
                
                <div
                  class="btn-group btn-group-toggle d-flex ng-untouched ng-pristine ng-valid"
                  ng-reflect-model="option1-1"
                  ng-reflect-name="feature1"
                  ngbradiogroup=""
                  role="group"
                >
                  
                  
                  
                  <label
                    class="btn-primary w-100 btn"
                    ng-reflect-ngb-tooltip="option 1-1 desc"
                    ngbbuttonlabel=""
                  >
                    
                    
                    <input
                      id="option1-1"
                      name="feature1"
                      ng-reflect-value="option1-1"
                      ngbbutton=""
                      type="radio"
                      value="option1-1"
                    />
                    
                    option 1-1
                  
                  </label>
                  
                  <label
                    class="btn-primary w-100 btn"
                    ng-reflect-ngb-tooltip="option 1-2 desc"
                    ngbbuttonlabel=""
                  >
                    
                    
                    <input
                      id="option1-2"
                      name="feature1"
                      ng-reflect-value="option1-2"
                      ngbbutton=""
                      type="radio"
                      value="option1-2"
                    />
                    
                    option 1-2
                  
                  </label>
                  
                  <label
                    class="btn-primary w-100 btn"
                    ng-reflect-ngb-tooltip="option 1-3 desc"
                    ngbbuttonlabel=""
                  >
                    
                    
                    <input
                      id="option1-3"
                      name="feature1"
                      ng-reflect-value="option1-3"
                      ngbbutton=""
                      type="radio"
                      value="option1-3"
                    />
                    
                    option 1-3
                  
                  </label>
                  
                  
                
                </div>
                
              
              </div>
              
            
            </div>
            <div
              class="form-group row"
            >
              
              
              <label
                class="option-label col-md-2"
                ng-reflect-ngb-tooltip="feature 2 desc"
              >
                feature 2
              </label>
              
              
              <div
                class="col-md-10"
              >
                
                
                <div
                  class="btn-group btn-group-toggle d-flex ng-untouched ng-pristine ng-valid"
                  ng-reflect-model="option2-1"
                  ng-reflect-name="feature2"
                  ngbradiogroup=""
                  role="group"
                >
                  
                  
                  
                  <label
                    class="btn-primary w-100 btn"
                    ng-reflect-ngb-tooltip="option 2-1 desc"
                    ngbbuttonlabel=""
                  >
                    
                    
                    <input
                      id="option2-1"
                      name="feature2"
                      ng-reflect-value="option2-1"
                      ngbbutton=""
                      type="radio"
                      value="option2-1"
                    />
                    
                    option 2-1
                  
                  </label>
                  
                  <label
                    class="btn-primary w-100 btn"
                    ng-reflect-ngb-tooltip="option 2-2 desc"
                    ngbbuttonlabel=""
                  >
                    
                    
                    <input
                      id="option2-2"
                      name="feature2"
                      ng-reflect-value="option2-2"
                      ngbbutton=""
                      type="radio"
                      value="option2-2"
                    />
                    
                    option 2-2
                  
                  </label>
                  
                  
                
                </div>
                
              
              </div>
              
            
            </div>
            
          
            
          
          </div>
          
        
          
          
          
          
        
          
      
        </div>
        
    
      </ngb-tabset>
      
      
      <div
        class="row"
      >
        
        
        <label
          class="option-label col-md-2"
        >
          Opponents
        </label>
        
        
        <div
          class="col-md-10"
        >
          
          
          <p-multiselect
            class="ui-inputwrapper-filled ng-untouched ng-pristine ng-valid"
            datakey="md5"
            maxselectedlabels="10"
            ng-reflect-data-key="md5"
            ng-reflect-max-selected-labels="10"
            ng-reflect-model=""
            ng-reflect-option-label="displayName"
            ng-reflect-options="[object Object]"
            optionlabel="displayName"
          >
            
          
            <div
              class="ui-multiselect ui-widget ui-state-default ui-corner-all"
              ng-reflect-ng-class="[object Object]"
            >
              
              
              <div
                class="ui-helper-hidden-accessible"
              >
                
                  
                <input
                  readonly="readonly"
                  type="text"
                />
                
              
              </div>
              
              
              <div
                class="ui-multiselect-label-container"
                title="Choose"
              >
                
                  
                <label
                  class="ui-multiselect-label ui-corner-all"
                >
                  Choose
                </label>
                
              
              </div>
              
              
              <div
                class="ui-multiselect-trigger ui-state-default ui-corner-right"
                ng-reflect-ng-class="[object Object]"
              >
                
                  
                <span
                  class="ui-clickable fa fa-fw fa-caret-down"
                  ng-reflect-klass="ui-clickable"
                  ng-reflect-ng-class="fa fa-fw fa-caret-down"
                />
                
              
              </div>
              
              
              <div
                class="ui-multiselect-panel ui-widget ui-widget-content ui-corner-all ui-shadow"
                ng-reflect-ng-class="ui-multiselect-panel ui-widget"
                style="display: none;"
              >
                
                  
                <div
                  class="ui-widget-header ui-corner-all ui-multiselect-header ui-helper-clearfix"
                  ng-reflect-klass="ui-widget-header ui-corner-all"
                  ng-reflect-ng-class="[object Object]"
                >
                  
                      
                  
                  <div
                    class="ui-chkbox ui-widget"
                  >
                    
                          
                    <div
                      class="ui-helper-hidden-accessible"
                    >
                      
                              
                      <input
                        readonly="readonly"
                        type="checkbox"
                      />
                      
                          
                    </div>
                    
                          
                    <div
                      class="ui-chkbox-box ui-widget ui-corner-all ui-state-default"
                      ng-reflect-klass="ui-chkbox-box ui-widget ui-cor"
                      ng-reflect-ng-class="[object Object]"
                    >
                      
                              
                      <span
                        class="ui-chkbox-icon ui-clickable"
                        ng-reflect-klass="ui-chkbox-icon ui-clickable"
                        ng-reflect-ng-class="[object Object]"
                      />
                      
                          
                    </div>
                    
                      
                  </div>
                  
                      
                  
                  <div
                    class="ui-multiselect-filter-container"
                  >
                    
                          
                    <input
                      class="ui-inputtext ui-widget ui-state-default ui-corner-all"
                      role="textbox"
                      type="text"
                    />
                    
                          
                    <span
                      class="fa fa-fw fa-search"
                    />
                    
                      
                  </div>
                  
                      
                  <a
                    class="ui-multiselect-close ui-corner-all"
                    href="#"
                  >
                    
                          
                    <span
                      class="fa fa-close"
                    />
                    
                      
                  </a>
                  
                  
                </div>
                
                  
                <div
                  class="ui-multiselect-items-wrapper"
                >
                  
                      
                  <ul
                    class="ui-multiselect-items ui-multiselect-list ui-widget-content ui-widget ui-corner-all ui-helper-reset"
                    style="max-height: 200px;"
                  >
                    
                          
                    
                    
                      
                  </ul>
                  
                  
                </div>
                
              
              </div>
              
          
            </div>
            
      
          </p-multiselect>
          
        
        </div>
        
      
      </div>
      
      
      <div
        class="row"
      >
        
        
        <div
          class="col-md-6"
        >
          
          
          <button
            class="btn btn-primary create-game-button w-100"
          >
            
            Create Game!
          
          </button>
          
        
        </div>
        
        
        <div
          class="col-md-6"
        >
          
          
          <button
            class="btn btn-info invite-friends-button w-100"
          >
            
            Invite friends to play
          
          </button>
          
        
        </div>
        
      
      </div>
      
    
    </form>
    
  
  </div>
  
</app-create-game>
`;
